[CM H1] Update Buildserver Documentation (#3707)
ElektraInitiative/libelektra: doc/BUILDSERVER.md

}
```

we can use any function that is declared in the shared library.

It is available in the [following repository](https://github.com/ElektraInitiative/jenkins-library).
For more detailed usage instruction or instructions on how to integrate this shared library into Jenkins, see its [README.md](https://github.com/ElektraInitiative/jenkins-library/blob/master/README.md).

### Jenkinsfiles

Jenkinsfiles describe what actions the build system should execute on which
build slave.
Currently Elektra uses four different files.

#### Jenkinsfile.daily

- [Jenkinsfile.daily](/scripts/jenkins/Jenkinsfile.daily) contains daily maintenance tasks, like cleaning up build servers.
- [Buildjob: libelektra-daily](https://build.libelektra.org/job/libelektra-daily/)
- [Jenkinsfile.daily](https://master.libelektra.org/scripts/jenkins/Jenkinsfile.daily)

#### Jenkinsfile.monthly

- [Jenkinsfile.monthly](/scripts/jenkins/Jenkinsfile.monthly) contains monthly triggering of libelektra job on master.
- [Buildjob: libelektra-monthly](https://build.libelektra.org/job/libelektra-monthly/)
- [Jenkinsfile.monthly](https://master.libelektra.org/scripts/jenkins/Jenkinsfile.monthly)

#### Jenkinsfile.release

> - Triggered manually with optional arguments and is for automated releasing a new version of Elektra.

Reviewer: ```suggestion
- Triggered manually with optional arguments. It is used for automated releasing a new version of Elektra.
```

For runs of the build job that are run in the master branch we also execute
deployment steps after all tests pass.

> We use it to build Debian and Fedora packages and move them into the repositories

Reviewer: ```suggestion
We use these steps to build Debian and Fedora packages and move them into the repositories
```

hosted on the community node.

Additionally we recompile the homepage and deploy it on the community node.

## Jenkinsfile.release (Release Pipeline)

### Release Builds

The `buildRelease` function runs various test suites and collects and archives debug
information. It also generates source-packages and packages for Debian based
distributions and Fedora.

The generated packages are additionally installed and tested on a clean
Docker image.

### Manual approval

> In order to verify the correct behavior and results of the previous steps and its

Reviewer: ```suggestion
In order to verify the correct behavior and results of the previous steps and their
```

generated artifacts, we use the [Input Step](https://www.jenkins.io/doc/pipeline/steps/pipeline-input-step/)
to pause the pipeline.
After manually verifying the correctness, the pipeline run can be resumed.

### Publishing

> The previously generated packages are published to its repositories, as described

Reviewer: ```suggestion
The previously generated packages are published to the repositories, as described
```

What does "its" refer to in "its repositories"?

above in _Jenkinsfile (Main CI Pipeline)_.

> The API documentation and source packages get published to its git repositories.

Reviewer: ```suggestion
The API documentation and source packages get published to the relevant git repositories.
```

Again what is this "it" you refer to?

- For Build Configuration you want to specify `by Jenkinsfile` and add the
  script path: `scripts/jenkins/Jenkinsfile`.

### Jenkins libelektra-release Configuration

The `libelektra-release` job is a normal pipeline job.

Most of the default settings can be used.

- Under General you need to enable `Permission to Copy Artifact` and set the field
  to `*`.
- For Build Configuration you want to specify `by Jenkinsfile` and add the
  script path: `scripts/jenkins/Jenkinsfile.release`.

### General Set-Up of a New Pipeline

We will cover how to set-up a pipeline that uses a different git repository than
this `libelektra` repository.

Most of the default settings can be used.

- If a private repository were to be used with a GitHub type job,
  a credentials pair of `elektrabot` must be selected to access this repository.
- For Build Configuration you want to specify `by Jenkinsfile` and add the
  script path to the Jenkinsfile.

> We first start by including the shared library as described above in _Jenkins Shared Library_.

Reviewer 2: Maybe move "Jenkins Shared Library" down here, it does not seem to be needed before?
